### DISCUSSION TITLE
Restrict user to own data in specific collection

### DISCUSSION THREAD

**Discussion by martin-braun** [directus/directus#14844] [Archive]
We have a license management system built up with directus. For simplicity let's say we have only two collections: "customers" and "licenses".

I've setup a role "managers". We want to want to assign one manager to one customer and all its licenses. The manager role can only read "customers" and edit "licenses". Obviously, this is not enough, because any manager can see all customers and all licenses of all customers. What we want is to restrict the a manager to have only access to one specific customer and to be able to see licenses that have been created by the same user.

Is there a way with directus?

**Comment by rijkvanzanten** [comment]
Sounds like that's possible through nested relationships on `$CURRENT_USER` in permissions 🙂 

> restrict the a manager to have only access to one specific customer and to be able to see licenses that have been created by the same user.

How would you chose what particular customer the "manager" has read access to? Is there going to be a field that says "these are the managers that have access"? When you say _created by the same user_, do you mean the manager, or the customer?

**Reply by martin-braun** [reply]
There is no relationship yet, but it would make sense to have a MANY to MANY relationship between customers and users to have it. When I say "created by the same user" I mean by the same manager. A manager should only see the licenses that he/she created. The former is possible, I simply set up a M:M relation from the customers collection to the system > directus_user collection, but I lack to see how I can restrict access to the manager in the backend, properly.

**Reply by domdus** [reply]
Just stumbled over this while dealing with very similar topic.

- Go to Settings -> Roles & Permissions
- Choose the role
- Choose collection `read` permission and click `Use custom`
<img width="163" alt="image" src="https://user-images.githubusercontent.com/67951828/183413162-62c0a733-5a4d-428e-b1c1-89f969ef4fef.png">

- Define rules a shown here:
<img width="468" alt="image" src="https://user-images.githubusercontent.com/67951828/183413412-236d7fbc-f8df-4243-9681-f7cd0b6e0c47.png">

Raw value:
`{
	"_and": [
		{
			"user_created": {
				"_eq": "$CURRENT_USER"
			}
		}
	]
}`

You may need to adapt with AND/OR to address other fields depending on your use case.
